xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp

```diff
@@ -576,6 +576,7 @@ void CVideoPlayerAudio::Process()
         msg.player = VideoPlayer_AUDIO;
         msg.cachetotal = cachetotal;
         msg.cachetime = cachetime;
+        msg.next_pts = m_messageQueue.GetTimeBack();
         msg.timestamp = m_audioClock;
```

Comment:
Long story short:

the MOV (Quicktime MP4 Movie) file I'm playing is muxed audio / video generated from segmented DASH stream. When I detect bandwith changes I add a new MOOV Header followed by the mp4 packets with the new quality / resolution. This currently leads to a stream change for both audio and video (regardless if only one has changed).

But regardless what comes in, I think it could be possible in kodi to handle it if we understand what happens.
0.5 secs are a lot - and they are always reproducable - so it's a good position to find the reason. I really want to understand the WHY. I have learned some things this night and will come back if I'll find anything.

If you want to support then you have to path the ffmpeg mov.c, the original version does not support multiple MOOV Atoms and the video looks ugly after 20 secs.